pkg/finder/podresource.go

```diff
@@ -0,0 +1,61 @@
+package finder
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	podresourcesapi "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+)
+
+// DevicesProvider knows how to provide the devices used by the given container
+type DevicesProvider interface {
+	GetDevices(podUID, containerName string) []*podresourcesapi.ContainerDevices
+	UpdateAllocatedDevices()
+}
+
+// PodsProvider knows how to provide the pods admitted by the node
+type PodsProvider interface {
+	GetPods() []*v1.Pod
+}
+
+// podResourcesServer implements PodResourcesListerServer
+type podResourcesServer struct {
+	podsProvider    PodsProvider
+	devicesProvider DevicesProvider
+}
+
+// NewPodResourcesServer returns a PodResourcesListerServer which lists pods provided by the PodsProvider
+// with device information provided by the DevicesProvider
+func NewPodResourcesServer(podsProvider PodsProvider, devicesProvider DevicesProvider) podresourcesapi.PodResourcesListerServer {
```

Comment:
could you please elaborate about why we need the server side here? Isn't RTE the consumer here?

Reply:
This file was removed as the implementation progressed, the commits have been squashed now